### PR TITLE
Wire up login

### PR DESCRIPTION
**Pull request by jpellizzari**
Gets the auth page to show up in development:
Screenshot from 2022-03-01 10-39-54

Most of this is incorherent tweaks to get things to render. I had started on a larger refactor (see `/hooks/featureflags.ts`), but cut it short in favor of getting others unblocked.

This also adds a central `fetch` mapping to lay the ground work for tests in the future.

Follow ups:
- [ ] #1563 
- [ ] #1564

@Callisto13 This is working for me via the `localhost:4567` server started via `npm start`, with `make cluster-dev` running in another terminal. I don't know what the state of the embedded files are on the server.

### CODE REVIEW

File: .proxyrc [hunk 1]
```diff
@@ -3,6 +3,6 @@
     "target": "http://localhost:9000/",
   }, 
   "/oauth2": {
-    "target": "http://localhost:9001/"
```
Comment (yitsushi) [inline .proxyrc:6]: Isn't this connected (somehow) to `containerPort: 9001` in `wego-app-dev.yaml`?

**Comment by AlinaGoaga** [comment]
Hey @jpellizzari! The FE code looks good to me and thanks for cleaning it up! Claudia kindly assisted me with trying to get this up and running on my computer but we couldn't get it to work and since there were quite a few files that I understand needed adjustments made, we ended up testing it on Claudia's computer. We've noticed a bug (when the flag is set to false you still have access to the sign_in page - it should redirect you to the first page of the app instead). I've not tested main but I can imagine the same happens there. Up to you if you want to fix it here or we can fix it in v1. Otherwise looking good. Thanks!

File: tools/wego-app-dev.yaml [hunk 2]
```diff
@@ -20,6 +20,10 @@ spec:
             - containerPort: 9001
               protocol: TCP
           imagePullPolicy: IfNotPresent
+          env:
+            - name: WEAVE_GITOPS_AUTH_ENABLED
+              value: "true"
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply tools/wego-app-dev.yaml]: I wasn't sure if we wanted to keep it, so I put it in it's own commit. I will drop it.

**Comment by jpellizzari** [comment]
> Hey @jpellizzari! The FE code looks good to me and thanks for cleaning it up! Claudia kindly assisted me with trying to get this up and running on my computer but we couldn't get it to work and since there were quite a few files that I understand needed adjustments made, we ended up testing it on Claudia's computer. We've noticed a bug (when the flag is set to false you still have access to the sign_in page - it should redirect you to the first page of the app instead). I've not tested main but I can imagine the same happens there. Up to you if you want to fix it here or we can fix it in v1. Otherwise looking good. Thanks!

@Callisto13 @AlinaGoaga Is this a blocker? Otherwise, I'm sure it will get worked out soonish in the normal course of dev work.

**Comment by AlinaGoaga** [comment]
@jpellizzari Nope, totally fine to look at it later ;)

File: .proxyrc (hunk 1 above)
Comment (jpellizzari) [reply .proxyrc:6]: Possibly, but this `proxy` is meant to be used on conjunction with `tilt`, so we use the port it listens on in the Tiltfile

**Pull request merged**